src/Nethermind/Nethermind.Consensus/Validators/TxValidator.cs

```diff
@@ -2,294 +2,450 @@
 // SPDX-License-Identifier: LGPL-3.0-only
 
 using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Collections.Generic;
 using Nethermind.Consensus.Messages;
 using Nethermind.Core;
-using Nethermind.Core.Crypto;
-using Nethermind.Core.Extensions;
 using Nethermind.Core.Specs;
+using Nethermind.TxPool;
+using Nethermind.Core.Crypto;
 using Nethermind.Crypto;
 using Nethermind.Evm;
 using Nethermind.Int256;
-using Nethermind.TxPool;
 
-namespace Nethermind.Consensus.Validators
+namespace Nethermind.Consensus.Validators;
+
+public sealed class TxValidator(ulong chainId) : ITxValidator
 {
-    public class TxValidator : ITxValidator
+    private readonly Dictionary<TxType, ITxValidator> _validators = new()
     {
-        private readonly ulong _chainIdValue;
-
-        public TxValidator(ulong chainId)
         {
-            _chainIdValue = chainId;
-        }
+            TxType.Legacy, new AllTxValidator([
+                IntrinsicGasTxValidator.Instance,
+                new LegacySignatureTxValidator(chainId),
+                ContractSizeTxValidator.Instance,
+                NonBlobFieldsTxValidator.Instance,
+            ])
+        },
+        {
+            TxType.AccessList, new AllTxValidator([
+                new ReleaseSpecTxValidator(spec => spec.IsEip2930Enabled),
```

Comment:
```suggestion
                new ReleaseSpecTxValidator(static spec => spec.IsEip2930Enabled),
```

Add to all registries.